File: lista_3/enigma.py
mensagem = input()
criterio = input()

copiam = []
for i in mensagem:
    copiam.append(i)
copiac = []
for j in criterio:
    copiac.append(j)

ok = 0
tamanho = len(copiac)
while len(copiam) >= tamanho:
    fatia = copiam[0:tamanho]
    Ih = 0
    for k in range(tamanho):
        if fatia[k] == copiac[k]:
            Ih += 1
            break
    if Ih == 0:
        ok += 1
    copiam.pop(0)
    
print(ok)

File: lista_3/enigma.rb
mensagem = gets.chomp.chars
criterio = gets.chomp.chars

possivel = 0
tamanho = criterio.length

while (mensagem.length >= criterio.length)

    particao = mensagem.slice(0, tamanho)
    
    indice = 0
    while (particao[indice] != criterio[indice] and indice < tamanho)
    	indice += 1
    end
    
    if (indice == tamanho)
    	possivel += 1
    end
    
    mensagem.delete_at(0)
 
end 
   
puts possivel
